File: IntentServiceStop/app/src/main/java/example/k/servicedemo/MainActivity.java
package example.k.servicedemo;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

//https://stackoverflow.com/questions/11258083/how-to-force-an-intentservice-to-stop-immediately-with-a-cancel-button-from-an-a
public class MainActivity extends AppCompatActivity implements ResultListner {
    public static final String ACTION_STOP = "stop";

    TextView textView;
    Button startService, stopService;
    MyBroadcastReceiver broadcastReceiver;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);

        startService = findViewById(R.id.btnstartService);

        stopService = findViewById(R.id.btnstopService);

        startService.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(MainActivity.this, MyIntentService.class);
                startService(intent);
            }
        });
        stopService.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                MyIntentService.shouldContinue = false;


            }
        });


    }

    @Override
    protected void onStart() {
        broadcastReceiver = new MyBroadcastReceiver();
        IntentFilter intentFilter = new IntentFilter();
        intentFilter.addAction(ACTION_STOP);

        registerReceiver(broadcastReceiver, intentFilter);
        super.onStart();
    }

    @Override
    protected void onStop() {
        unregisterReceiver(broadcastReceiver);
        super.onStop();
    }

    public void onDestroy() {
        unregisterReceiver(broadcastReceiver);
        super.onDestroy();

    }


    @Override
    public void Result(String result) {
        textView.setText(result);

    }
}

File: app/src/main/java/services/localbroadcastupdateuisecond/MainActivity.kt
package services.localbroadcastupdateuisecond

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.os.Handler
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import java.util.*

class MainActivity : AppCompatActivity() {
    private lateinit var textview: TextView
    private lateinit var startService: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textview = findViewById(R.id.textview)
        //startService = findViewById(R.id.startService)
        val handler = Handler()
        var timer=Timer()
    var count =0

        val timerTask = object : TimerTask() {
            override fun run() {
                handler.post(object : Runnable {
                    override fun run() {
                        val intent = Intent(applicationContext, MyIntentService::class.java)
                        startService(intent)
                        Toast.makeText(applicationContext,"call"+ count++,Toast.LENGTH_LONG).show()
                    }

                })

            }

        }
        timer.schedule(timerTask, 0, 10000)
      /*  startService.setOnClickListener {
            val intentService = Intent(this, MyIntentService::class.java)
            startService(intentService)
        }*/
    }

    override fun onStart() {
        super.onStart()
        val intentFilter = IntentFilter()
        intentFilter.addAction("services.localbroadcastupdateuisecond")
        LocalBroadcastManager.getInstance(this).registerReceiver(broadcastReceiver, intentFilter)
    }

    override fun onStop() {
        super.onStop()
        LocalBroadcastManager.getInstance(this).unregisterReceiver(broadcastReceiver)
    }

    private val broadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            val data = intent?.getStringExtra("Datapassed")
            textview.text = data.toString()
        }

    }
}